coordinator/internal/transitengine/transitengine_test.go

```diff
@@ -0,0 +1,167 @@
+			Plaintext:      "vpIh",
+			name:           "wrong/url",
+			AssociatedData: "vpIh",
+			expStatus:      404,
+		},
+		"f_bad-request-nob64plain": {
+			Plaintext:      "thi$$hoU_ld+*notwork",
+			name:           "autounseal",
+			AssociatedData: "vpIh",
+			expStatus:      400,
+		},
+		"f_bad-request-nob64AD": {
+			Plaintext:      "HribV+LFZspJpAauFf643A1HKbj1VlQWVhAKFDJqdZg=",
+			name:           "autounseal",
+			AssociatedData: "thi$$hoU_ld+*notwork",
+			expStatus:      400,
+		},
+		"f_neg_version": {
+			Plaintext:      "lT3rQGMlxq680DdSKfIYYcfyCfMnP9ikxaO5b0mGRKRl4qNL3W9xkSW3QmaMwozCRfNMZhhDCbYokn6KEiGotlVInKt66QjBgXR2Nk9hIcez0LYt8W5pxD0lwTxC",
+			name:           "thi$$hoU_ld+*work",
+			AssociatedData: "HribV+LFZspJpAauFf643A1HKbj1VlQWVhAKFDJqdZg=",
+			Version:        -2000,
+			expStatus:      400,
+		},
+	}
+
+	t.Run("encrypt-decrypt handler", func(t *testing.T) {
+		fakeStateAuthority, _ := NewFakeSeedEngineAuthority()
+		mux := NewTransitEngineAPI(fakeStateAuthority, slog.Default())
+
+		for testname, entry := range mocks {
+			t.Run(testname, func(t *testing.T) {
+				var ciphertext, receivedPlaintext string
+				t.Run("encryption request handling", func(t *testing.T) {
+					require := require.New(t)
+					jsonBody, err := json.Marshal(entry)
+					require.NoError(err)
+					req := httptest.NewRequest(http.MethodPut, "/v1/transit/encrypt/"+entry.name, bytes.NewReader(jsonBody))
+					req.Header.Set("Content-Type", "application/json")
+					rec := httptest.NewRecorder()
+					mux.ServeHTTP(rec, req)
+					res := rec.Result()
+					require.Equal(entry.expStatus, res.StatusCode)
+
+					if res.StatusCode != http.StatusOK {
+						return
+					}
+					encRespBody, err := parseResponseBody(res)
+					require.NoError(err)
+					data, _ := encRespBody["data"].(map[string]any)
+					ciphertext, _ = data["ciphertext"].(string)
+				})
+
+				t.Run("decryption request handling", func(t *testing.T) {
+					require := require.New(t)
+					decryptReqBody, err := json.Marshal(
+						map[string]string{
+							"ciphertext":      ciphertext,
+							"associated_data": entry.AssociatedData,
+						})
+					require.NoError(err)
+					decryptReq := httptest.NewRequest(http.MethodPut, "/v1/transit/decrypt/"+entry.name, bytes.NewReader(decryptReqBody))
+					decryptReq.Header.Set("Content-Type", "application/json")
+					rec := httptest.NewRecorder()
+					mux.ServeHTTP(rec, decryptReq)
+					res := rec.Result()
+					require.Equal(entry.expStatus, res.StatusCode)
+					if res.StatusCode != http.StatusOK {
+						return
+					}
+					decRespBody, err := parseResponseBody(res)
+					require.NoError(err)
+					data, _ := decRespBody["data"].(map[string]any)
+					receivedPlaintext, _ = data["plaintext"].(string)
+					require.Equal(entry.Plaintext, receivedPlaintext, "Unexpected received plaintext after cycling handler functions")
+				})
+			})
+		}
+	})
+}
+
+func parseResponseBody(res *http.Response) (map[string]any, error) {
```

Comment:
If I were to inline this function, it would boil down to

```diff
- decRespBody, err := parseResponseBody(res)
- require.NoError(err)
+ defer res.Body.Close()
+ respBody, err := io.ReadAll(res.Body)
+ require.NoError(err)
+ var decRespBody map[string]any
+ require.NoError(json.Unmarshal(respBody, &decRespBody))
```

I'd prefer the inlined version because it does not interrupt my reading flow (need to figure out what `parseResponseBody` does).

Reply:
Actually, you can make it even simpler with 

```go
require.NoError(json.NewDecoder(respBody).Decode(&decRespBody))
```